### ISSUE TITLE
4 million domains missing

### ISSUE THREAD

**Issue by libBletchley** [themusicgod1/cloudflare-tor#1]
I'm wondering why the `cloudflare-list.txt` is under 1500 sites, while [this list](https://github.com/pirate/sites-using-cloudflare) is over 4 million.  Is this the human/manual data entry project of Shiromarike?  Should these projects be merged?

**Comment by themusicgod1** [comment]
I've never heard of @shiromarleke & I've reached out to them for more details.  This list is made by human hands.  I've mentioned the other list in the README now.

> The end-game should be to have a list of sites that use the CF proxy, as well as the subset of those that DoS attack the Tor community.

Not a bad idea.

**Comment by themusicgod1** [comment]
Turns out it is, in fact the continuation of @shiromarieke's & @CryptopartyBLN, project.

**Comment by shiromarieke** [comment]
Hey. So, yes this was a project I started with dawning_sun from CryptoParty Berlin (not the whole CryptoParty team, most had nothing to do with this). Ofc it's not exhaustive at all, it was never supposed to be. It was more a campaign item (show people care as there was a lot of people contributing, we generally campaigned using #dontblocktor on twitter, which I initially launched for something else but then was used for this), and it was also meant to see which blocks actually impact tor users the most. I have to say the Clownflare problem, even if still present, got much better since the time I started campaigning. We used to get stuck in endless captcha loops, they were still using google's old recptcha, which was absolutely impossible to read, and Clownflare now offers to their users the option to unblock tor. Happy to see people still care about documenting this BS though :)  I'm not working on this anymore but happy to answer any questions you have.

**Comment by libBletchley** [comment]
I think it would be useful if a searx instance were to use this list and either purge those sites from search results, or fold them and put a hazard triangle w/warning msg, or rank them down to page 10 or something.

**Comment by themusicgod1** [comment]
I think the 'searx instance' is in general a good idea but beyond the scope of this repository/project.  It's something that could be built on top of, and using this list, but not something that we should have here.  Similarly, a mastodon instance for example could make it so that every time a user tries to link to a domain on this list it prompts them with a warning message, or auto-marks their toot with a CW.  But again - that's the kind of thing that you can get *other* projects to do, now that this list exists and is ideally kept updated.

 [ ] Having a list separated into 'openly hostile against tor' / 'merely MiTM' 

however is something human hands can do, that's what this thread is going to become unless you dissent.

**Comment by themusicgod1** [comment]
maybe there's a 3rd category worth marking here - 
1) blocks tor entirely (blacklist)
2) exposes tor users to CAPTCHA (MiTM+semi-effective block)
3) whitelist tor users (still MiTM)

**Comment by libBletchley** [comment]
For category 1, there are websites that go beyond tor and block anyone using an anonymizing service.  They try to detect VPNs, known proxies, non-residential IPs, etc.  So you might say it's general blocking based on IP reputation.  Sometimes it's based on geography.  Like a 2nd-hand shop blocking visitors from outside the country (even if they are locals on vacation or whatever).

Nastier are the shops that let you shop on their site, then you discover when your package is not delivered that they discriminated against your IP after you placed an order, and they generally don't even tell you why the order/service was cancelled, or even that it was cancelled.  It's the worst kind of DoS attack in this category because it wastes your time and possibly ruins your plans.  (BTW, walmart.com does what I just described)

Some shops block you because your IP does not appear to be local, but then if you VPN to a local host they block you because your IP is non-residential (from a data center).

For category 2, a captcha may or may not be triggered based on IP.  And if it is based on IP, it's usually a tor IP, but exceptionally there are sites like sears.com that will captcha international visitors.

For category 3, of course whitelisted cloudflare is what you have in mind, but any CDN is functionally in this category.  Although CloudFlare is the most alarming and least ethical of them.  It might be useful to know who the MitM is.  People should know who they are sharing their traffic with.

Maybe it's crazy talk or getting carried away, but ideally a db would capture all those details, which could then be used by search engines and browsers to help people avoid time-waste and privacy-breeching shenanigans.

There should be a special hell for government websites that do any of the above.

cat 1:
 * healthcare.gov
 * fbi.gov
 * fcc.gov
 * usa.gov
 * irs.gov

cat 2:
 * gpo.gov
 * fccid.io (not sure it's government run)

Category 1 gov sites should be seen as a 4th amendment violation (giving up my IP address is an unreasonable search IMO).  They don't even disclose that tor is the problem, so it's an unjustified denial of service - which is a bit off for healthcare and irs who publish essential information.  And worse, proxy and archive sites will say "we don't have permission to copy this site" or something to that effect.

And if it's a government site where you can submit feedback, then the block is effectively a 1st amendment violation.

If a bank or accountant uses a CDN, I suspect it could be treated as a [Gramm–Leach–Bliley Act](https://en.wikipedia.org/wiki/Gramm%E2%80%93Leach%E2%80%93Bliley_Act) violation, which stipulates that banks and accountants must safeguard your personal info from 3rd parties (e.g. they must use encryption when emailing it).

However the list is organized, I think how the data is used should be considered, whether it's used for legal actions or smarter tools and search engines.

**Comment by themusicgod1** [comment]
* I agree that a this splitting list up means it's really becoming a table - especially with tags a database makes some sense technically.  And I agree that if anyone actually uses this list we will work with them on making sure formatting works for as many people as possible.

On the other hand, at the end of the day lists, and input into lists, is done by human beings including this one. While there's millions of potential domains missing, and we may get to the point where a database is the only feasible way of dealing with this but for the moment there's maybe a few dozen people who have actually used this list that I'm aware of and none of them are sending me pull requests. I think at this stage it'd be premature optimization, especially since it'd be easy to make a database with this data.  However

 - [ ] make working scripts for a sqlite3 database using this data

is a task we could do.

* On the subset of category 1 who are just plain assholes to their customers like this, it does seem that they are out there -- but maybe instead of splitting category 1 we can add this sort of thing as metadata/flags.  That way we have a simpler set of categories to work with, while at the same time highlighting exactly what nefarious behaviour the category 1 crowd is actually doing. Thoughts?

In the meanwhile , this suggests another task:

 - [ ]  make new tag for asshole companies who break only *after* you try to place an order

**Comment by libBletchley** [comment]
> On the subset of category 1 who are just plain assholes to their customers like this, it does seem that they are out there -- but maybe instead of splitting category 1 we can add this sort of thing as metadata/flags. That way we have a simpler set of categories to work with, while at the same time highlighting exactly what nefarious behaviour the category 1 crowd is actually doing. Thoughts?

Indeed, I'm thinking an attribute that indicates whether the service denial is *deferred* or *instant* would be useful.  Deferred denials are often the nastiest, but in some cases (wikis) more favorable than instant blocks.  E.g.
 * (deferred denial favorable) A wiki page allows the world to view it, but when it comes to editing the page will block edits and registration.  This is less harmful than an instant block, because read access is useful.
 * (deferred denial unfavorable) JP Morgan will accept tor connections just to the point of login credential entry, and then not only disable their account at that moment but block their bank card, and put it in a non-unlockable state that requires the customer to request a new account number.  No banker or security admin can reverse the action and re-activate the account with the same number.  So the bank is actually opening itself up to attacks over tor (compromising security) just to be able to catalog and disservice legit customers who value their privacy.  It's the most obnoxious case of a deferred denial, and it's also the most important information to publish because the damage is done before a user knows to avoid it, and it's very inconvenient to fix.

Anyway, I'm just brainstorming what's needed in the end.  It may be unrealistic to try to track some of this.  Whether a deferred DoS is more or less harmful could be left out - ppl can probably figure it out.  I would envision some of these attributes:

- [ ] service denial is instant/deferred
- [ ] MitM imposed (yes/no) and if _yes_, who (cloudflare, perimeterX, etc)
- [ ] DoS basis (anonymous IP, IP region, data center IP, residential IP, etc) if anonymous IP, is it based on tor, i2p, known vpn/proxy, etc.
- [ ] Anti-human features (CAPTCHAs)
- [ ] nature of site (commercial [finance,retail goods,other], non-commercial [gov,education,charity])
- [ ] call for action (boycott, discourage donations, petition, legal action) Could link to an article or petition.

Would also perhaps be useful to track harmful associations/partnerships.  E.g. if you use Tutanota.de, you may be directed to uservoice.com, a cloudflare site -- and therefore would be encouraged to act against tutanota.  As an example of hypocrisy, the Tor project accepts BTC donations over BitPay, a cloudflare site.  The concept of a conditional donation seems appealing.  Someone could start a campaign to offer to donate X amount to Tor Project if they stop using cloudflared sites for any of their donations or services.

**Comment by libBletchley** [comment]
> Also your suggestion of actual law that could be brought to wield against cloudflare/etc suggests the need for another list, that I'm not sure yet if it's in the context of this repository or not(why not?)

In the case of accountants using CF, it would actually be the accountant who would be possibly breaking the law by sharing.  What might be useful is to name the regulatory authorities for the offending org.  E.g. if it's a bank or accountant that's breaking the law, the CFPB would be mentioned.  And actually there need not be a law broken.. CFPB will handle complaints in general about customer service if I understand correctly.  So to simply have a field in the db naming the relevant regulatory authority might be useful.

**Comment by libBletchley** [comment]
This FOSDEM talk could give good advice for getting your project funding, if needed:

https://fosdem.org/2018/schedule/event/eufunding/

If you're not at FOSDEM, they'll tape it and publish it later.

**Comment by themusicgod1** [comment]
Noted. I have all the funding i'll ever need for anything, tbh - asking for more funding would be kind of ridiculous at this point.  Thank you for the suggestion though, I do appreciate it.  Funding isn't what I need to do more interesting things - it's developer-hours and debian-developer-hours.  

That said, I can't leave my country/travel to europe & will probably get permalocked in a cage offline before I'm able to do *too* much more on this project, so if y'all think there's something to be made here you're going to probably have to start getting used to carrying on without me now.

**Comment by usingtorisnotacrime** [comment]
Let's do this. I'm tired of solving Captchas (AmA Tor user) and I also dislike "HTTPS MITM by Cloudflare" thing.

@themusicgod1 Plz check your email (but don't link it to me) ;)

**Comment by usingtorisnotacrime** [comment]
@libBletchley I'll add your suggestion to my searx instance.

**Comment by libBletchley** [comment]
> I'll add your suggestion to my searx instance.

That's good news.  Whenever it's ready to use, please let us know which instance it is.

Thanks!

**Comment by usingtorisnotacrime** [comment]
@libBletchley
It depends on this repository's owner. There's some work which needs to be done.
Are you a member of this repository(have write access)?

**Comment by usingtorisnotacrime** [comment]
I need a list of "base domains". Current list is hard to parse.

Current list have these lines:
ankisrs.net [FLOSSp] <= After yesterday complained
act.oxfam.org (oxfam.org seems ok)

I'm expecting: (base domains without comment)
ankisrs.net
oxfam.org

---
Why I said "base domain"?
Because Cloudflare users have to switch their DNS to cloudflare DNS server while sign-up.

So if oxfam.org is using Cloudflare, some.oxfam.org is also under Cloudflare control.

**Comment by usingtorisnotacrime** [comment]
...and done. Contains only base domain, so it's easy to use.
Feel free to merge it.
https://paste.ee/p/gO2R5

FYI crimeflare.com is sometimes not accessible.
Mirror: https://crimeflare.danwin1210.me

**Comment by themusicgod1** [comment]
re: 
> Feel free to merge it.
> https://paste.ee/p/gO2R5
can you pls make a PR for that?  paste.ee is inaccessible

**Comment by themusicgod1** [comment]
I mean that's a nice-to-have.
what would be even better would be a link to the sear.x instance

**Comment by usingtorisnotacrime** [comment]
Github website's editor form is broken. I can't copy & paste.
@libBletchley Maybe you can make a PR.

@themusicgod1 Did you check your email? I've sent searx URL (try it with Firefox - I'm blocking spyware aka Chrome to promote open source software)

**Comment by usingtorisnotacrime** [comment]
Current latest(I'm using this one). Ignore old URL(gO2R5).
https://paste.ee/p/o4KCo

**Comment by themusicgod1** [comment]
@ghost http 502

**Comment by unstableusername** [comment]
@themusicgod1 "with Firefox". The webpage don't serve spyware aka Chrome.

**Comment by themusicgod1** [comment]
@libBletchley there's now suggested tags in the instructions.  Now it's just a matter of someone going through, reading the comments and tagging where appropriate.  
I guess thinking about it 
NEEDSREVIEWp could be another one for ones that haven't been reviewed since the last change of format.

**Comment by libBletchley** [comment]
Now it seems use of CloudFlare DNS service is becoming controversial:

https://torrentfreak.com/cloudflare-and-riaa-agree-on-tailored-site-blocking-process-180501/

Just wanted to mention this so it's not overlooked (assuming it's in scope).. I still have to catch up w/this project.
